manual/src/cmds/intf-c.etex

```diff
@@ -476,48 +481,71 @@ gives a more complete example using "Custom_tag" blocks.
 /* Create an OCaml value encapsulating the pointer p */
 static value val_of_typtr(ty * p)
 {
-  value v = caml_alloc(1, Abstract_tag);
+  CAMLParam0();
+  CAMLlocal1(v);
+  v = caml_alloc(1, Abstract_tag);
+  // Data_abstract_val provides a pointer to the data area of the block
   *((ty **) Data_abstract_val(v)) = p;
-  return v;
+  CAMLreturn(v);
 }
 
 /* Extract the pointer encapsulated in the given OCaml value */
 static ty * typtr_of_val(value v)
 {
-  return *((ty **) Data_abstract_val(v));
+  CAMLparam1(v);
+  CAMLreturnT(ty *, *((ty **) Data_abstract_val(v)));
```

Comment:
I wouldn't register a GC root here, it is just a wrapper around an accessor function.